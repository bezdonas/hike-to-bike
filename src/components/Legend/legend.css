.legend-block {
  text-align: left;
  position: absolute;
  bottom: 20px;
  left: 20px;
}

.legend-block .leaflet-marker-icon {
  display: inline-block;
  vertical-align: middle;
  position: static;
  width: 25px;
  height: 41px;
}

.legend-block .note {
  color: rgba(0, 0, 0, 0.8);
  font-size: 12px;
}

@media (max-width: 700px) {
  .legend-block {
    position: fixed;
    background: rgba(250, 250, 250, 0.4);
  }
  .legend-block h3 {
    display: none;
  }
}
