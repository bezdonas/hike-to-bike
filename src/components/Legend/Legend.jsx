import { h, Component } from 'preact';
import './legend.css';

export default class Legend extends Component {
  render() {
    return (
      <div className="legend-block">
        <h3>Legend</h3>
        <p>
          <img
            src="https://unpkg.com/leaflet@1.3.4/dist/images/marker-icon-2x.png"
            class="leaflet-marker-icon"
          />{' '}
          &mdash; Healthy station
        </p>
        <p>
          <img
            src="https://unpkg.com/leaflet@1.3.4/dist/images/marker-icon-2x.png"
            class="leaflet-marker-icon"
            style="opacity: 0.5"
          />{' '}
          &mdash; Near-empty station
        </p>
        <span class="note">Markers are clickable for details</span>
      </div>
    );
  }
}
