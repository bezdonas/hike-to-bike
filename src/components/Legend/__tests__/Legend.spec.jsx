import { h, preact } from 'preact';
import render from 'preact-render-to-json';
import Legend from '../Legend.jsx';

describe('Legend', () => {
  it('renders as expected and matches snapshot', () => {
    expect(render(<Legend />)).toMatchSnapshot();
  });
});
