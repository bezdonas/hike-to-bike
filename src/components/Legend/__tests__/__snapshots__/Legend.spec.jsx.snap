// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend renders as expected and matches snapshot 1`] = `
<div
  class="legend-block"
>
  <h3>
    Legend
  </h3>
  <p>
    <img
      class="leaflet-marker-icon"
      src="https://unpkg.com/leaflet@1.3.4/dist/images/marker-icon-2x.png"
    />
     — Healthy station
  </p>
  <p>
    <img
      class="leaflet-marker-icon"
      src="https://unpkg.com/leaflet@1.3.4/dist/images/marker-icon-2x.png"
      style="opacity: 0.5"
    />
     — Near-empty station
  </p>
  <span
    class="note"
  >
    Markers are clickable for details
  </span>
</div>
`;
